### ISSUE TITLE
StarSpace notebook

### ISSUE THREAD

**Issue by SkBlaz** [vpodpecan/representation_learning#18]
It appears it does not automatically clone the repo:
![image](https://user-images.githubusercontent.com/10035780/101246344-fee08c80-3712-11eb-82f1-ddd3fcde8607.png)

There are two rather troubling settings, I suggest you address:

```
!./StarSpace/starspace train \
  -trainFile "data/ag_news_csv/train.pp" \
  -model "data/ag_news_csv/model" \
  -initRandSd 0.01 \
  -adagrad false \
  -ngrams 1 \
  -lr 0.01 \
  -epoch 5 \
  -thread 20 \
  -dim 10 \
  -negSearchLimit 5 \
  -trainMode 0 \
  -label "__label__" \
  -similarity "dot" \
  -verbose false
```

epoch = 5 is *way too small* this probably doesn't even converge. I suggest a value of at least 100 or more for this. The learning rate could also be a bit lower I reckon, e.g., `0.005`. Finally the `dim` is way too low. The standard value for this is `128`, please try that perhaps. I'm confident this will return better results.

**Comment by vpodpecan** [comment]
Sorry, but you are mistaken. I tried the settings you suggested (epochs=100, dim=128,lr=0.005) and I got **worse** results than with the default above:
hit@1: 0.451842 hit@10: 1 hit@20: 1 hit@50: 1 mean ranks : 1.76 Total examples : 7600
Try it if you don't belive :)

**Comment by vpodpecan** [comment]
Also, I just saw that in the paper authors explicitely state that the number of dimensions was set to 10 for the text classification task on AG news. So the parameters in the example above are the same as used in the paper.

**Comment by SkBlaz** [comment]
And the data is the same 100%?

**Comment by vpodpecan** [comment]
Yes, the AG news dataset 120K/7.6K examples.

**Comment by SkBlaz** [comment]
Are you 100% sure though? The original dataset does not seem available for example: https://di.unipi.it/%C5%93gulli/AG_corpus_of_news_articles.html

![image](https://user-images.githubusercontent.com/10035780/101906972-1b207580-3bba-11eb-86be-5dd2c02c2c76.png)

I'm very sceptical about these >90% not being at least approximately in the range though.

**Comment by vpodpecan** [comment]
The dataset used in the StarSpace example is hosted by FB so it remains available: https://dl.fbaipublicfiles.com/starspace/ag_news_csv.tar.gz
If you extract it you will see that this is the Zhan and LeCun's subset of the whole AG corpus (http://groups.di.unipi.it/~gulli/AG_corpus_of_news_articles.html) as reported in the paper. So there is absolutely no doubt whether the data is the same or not.

**Comment by SkBlaz** [comment]
Alright. Finally, is the preprocessing the same?

**Comment by vpodpecan** [comment]
Preprocessing is not detailed in the paper, except that dim=10. But even substantial differences in text preprocessing could not result in such discrepancy.

**Comment by SkBlaz** [comment]
I would add that as a potential interpretation of what might have gone
wrong, otherwise this still looks somewhat odd to a newcomer.
